hi-ml-cpath/src/health_cpath/models/encoders.py

```diff
@@ -62,18 +62,22 @@ class ImageNetEncoder(TileEncoder):
     """Feature extractor pretrained for classification on ImageNet"""
 
     def __init__(self, feature_extraction_model: Callable[..., nn.Module],
-                 tile_size: int, n_channels: int = 3) -> None:
+                 tile_size: int, n_channels: int = 3, apply_imagenet_preprocessing: bool = True) -> None:
         """
         :param feature_extraction_model: A function accepting a `pretrained` keyword argument that
         returns a classifier pretrained on ImageNet, such as the ones from `torchvision.models.*`.
         :param tile_size: Tile width/height, in pixels.
         :param n_channels: Number of channels in the tile (default=3).
+        :param apply_imagenet_preprocessing: Whether to apply ImageNet preprocessing to the input.
         """
         self.create_feature_extractor_fn = feature_extraction_model
+        self.apply_imagenet_preprocessing = apply_imagenet_preprocessing
         super().__init__(tile_size=tile_size, n_channels=n_channels)
 
     def _get_preprocessing(self) -> Callable:
-        return get_imagenet_preprocessing()
+        if self.apply_imagenet_preprocessing:
+            return get_imagenet_preprocessing()
+        return super()._get_preprocessing()
```

Comment:
this risks to lead to errors - the _get_preprocessing of the super class is ignored if the flag is true. I know the method is basically empty at the moment but we could change with a composition or with and an if/else?